gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -1072,6 +1625,67 @@ def validate(self, data):
 
         return data
 
+    def validate_to_publish(self, data):
+        """
+            Second step to validate the JSON data.
+            This validation is done before a record is published.
+            There are mandatory fields to publish a record:
+                - locus (validated in the first validation step)
+                - disease
+                - genotype/allelic requirement
+                - mutations consequence
+                - panel(s)
+                - confidence
+                - publication(s)
+            
+            data (CurationData obj): data to be validated
+        """
+
+        json_data = data.json_data
+        missing_data = []
+
+        # Check if G2P record (LGD) is already published
+        try:
+            lgd_obj = LocusGenotypeDisease.objects.get(
+                locus__name = json_data["locus"],
+                genotype__value = json_data["allelic_requirement"],
+                disease__name = json_data["disease"]["disease_name"]
+            )
+
+            raise serializers.ValidationError({
+                "message": "Found another record with same locus, genotype and disease",
+                "Please check G2P record": lgd_obj.stable_id.stable_id
+            })
+
+        except LocusGenotypeDisease.DoesNotExist:
+            if len(json_data["disease"]) == 0:
+                missing_data.append("disease")
+            
+            if len(json_data["confidence"]) == 0:
+                missing_data.append("confidence")
+            
+            if len(json_data["publications"]) == 0:
+                missing_data.append("publication")
+            
+            if not json_data["panels"]:
+                missing_data.append("panel")
+            
+            if json_data["allelic_requirement"] == "":
+                missing_data.append("allelic_requirement")
+
+            if missing_data:
+                raise serializers.ValidationError({"message" : f"The following mandatory fields are missing: {missing_data}"})
+
+            # Check if data is stored in G2P
+            # Locus - we only accept locus already stored in G2P
+            try:
+                locus_obj = Locus.objects.get(name=json_data["locus"])
+            except Locus.DoesNotExist:
+                raise serializers.ValidationError({"message" : f"Invalid locus {json_data["locus"]}"})
```

Comment:
Running this in my local is giving the error "SyntaxError: f-string: unmatched '['"
After changing "locus"(surrounded by double quotes) to 'locus' (surrounded by single quotes) fixed the issue
Please check if you are facing similar issue in your local